x/upgrade/keeper/keeper.go

```diff
@@ -325,23 +325,35 @@ func (k Keeper) IsSkipHeight(height int64) bool {
 	return k.skipUpgradeHeights[height]
 }
 
-// DumpUpgradeInfoToDisk writes upgrade information to UpgradeInfoFileName.
+// DumpUpgradeInfoToDisk writes upgrade information to UpgradeInfoFileName. The function
+// doesn't save the `Plan.Info` data, hence it won't support auto download functionality
+// by cosmvisor.
+// NOTE: this function will be update in the next release.
 func (k Keeper) DumpUpgradeInfoToDisk(height int64, name string) error {
+	return k.DumpUpgradeInfoWithInfoToDisk(height, name, "")
+}
+
+// Deprecated: DumpUpgradeInfoWithInfoToDisk writes upgrade information to UpgradeInfoFileName.
+// `info` should be provided and contain Plan.Info data in order to support
+// auto download functionality by cosmovisor and other tools using upgarde-info.json
+// (GetUpgradeInfoPath()) file.
+func (k Keeper) DumpUpgradeInfoWithInfoToDisk(height int64, name string, info string) error {
```

Comment:
I hate this kind of names.... but couldn't find better one

Reply:
I don't like it either lol. What about `*WithPlan`? Also, why is a new function being introduced and marked as deprecated?

Reply:
The whole thing is a Plan (name, height, info...).
In original solution I overloaded the `DumpUpgradeInfoToDisk` using additional vararg (that made it backward compatible). I agree with Aaron, that overloading doesn't smell good. see: https://github.com/cosmos/cosmos-sdk/pull/10532#discussion_r749722224